log4j-core/src/main/java/org/apache/logging/log4j/core/config/plugins/processor/PluginProcessor.java

```diff
@@ -107,6 +123,93 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
+                                    == 1 // It is a weird pattern to not have public setter
+                    ) {
+
+                        Types typeUtils = processingEnv.getTypeUtils();
+
+                        boolean followsNamePattern = methodName.equals(
+                                        String.format("set%s", expectedFieldNameInASetter(fieldName)))
+                                || methodName.equals(String.format("with%s", expectedFieldNameInASetter(fieldName)));
+
+                        // Check if method is public
+                        boolean isPublicMethod = methodElement.getModifiers().contains(Modifier.PUBLIC);
+
+                        // Check if the return type of the method element is Assignable.
+                        // Assuming it is a builder class the type of it should be assignable to its parent
+                        boolean checkForAssignable = typeUtils.isAssignable(
+                                methodElement.getReturnType(),
+                                methodElement.getEnclosingElement().asType());
+
+                        boolean foundPublicSetter = followsNamePattern && checkForAssignable && isPublicMethod;
+                        if (foundPublicSetter) {
+                            // Hurray we found a public setter for the field!
+                            return;
+                        }
+                    }
+                }
+            }
+            // If the setter was not found generate a compiler warning.
+            processingEnv
+                    .getMessager()
+                    .printMessage(
+                            Diagnostic.Kind.ERROR,
+                            String.format(
+                                    "The field `%s` does not have a public setter, Note that @SuppressWarnings(\"%s\"), can be used on the field to suppress the compilation error. ",
+                                    fieldName, SUPPRESS_WARNING_PUBLIC_SETTER_STRING),
+                            element);
+        }
+    }
+
+    /**
+     *  Helper method to get the expected Method name in a field.
+     *  For example if the field name is 'isopen', then the expected setter would be 'setOpen' or 'withOpen'
+     *  This method is supposed to return the capitalized 'Open', fieldName which is expected in the setter.
+     * @param fieldName who's setter we are checking.
+     * @return The expected fieldName that will come after withxxxx or setxxxx
+     */
+    public static String expectedFieldNameInASetter(String fieldName) {
+        if (fieldName.startsWith("is")) fieldName = fieldName.substring(2);
+
+        if (StringUtils.isEmpty(fieldName)) return ""; // Return empty string if the field is a blank String.
+
+        return String.format(
+                "%s%s",
+                fieldName.substring(0, 1).toUpperCase(Locale.ROOT),
+                fieldName.substring(1)); // we do not need to Lowercase here examples like field name 'ignoreExceptions'
```

Comment:
```suggestion
        return fieldName.isEmpty() ? fieldName : Character.toUpperCase(fieldName.charAt(0)) + fieldName.substring(1);
```

In this case, we don't need `StringUtils.isEmpty()`, since the argument is certainly non null. In the cases we need it, we have a copy in `o.a.l.l.util.Strings.isEmpty()`.

Reply:
Changed as per suggestion 😄